modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -3201,6 +3210,18 @@ private void _validate(
 		return null;
 	}
 
+	private void _validateSubscriptionCycles(
+			long subscriptionCycles, String subscriptionCyclesKey)
+		throws PortalException {
+
+		if (subscriptionCycles < 0) {
+			throw new CPDefinitionMaxSubscriptionCyclesException(
```

Comment:
please throw a CPDefinitionDeliveryMaxSubscriptionCycles for deliverySubscriptionCycles case

Reply:
Changed it, should be good now. Thank you